File: src/main/java/com/github/ioannuwu/errorlens/data/defaultsettings/DefaultSettingsList.java
package com.github.ioannuwu.errorlens.data.defaultsettings;

import com.github.ioannuwu.errorlens.data.SettingsState;

import java.awt.Color;
import java.util.List;

public class DefaultSettingsList {

    public static final int NUMBER_OF_WHITESPACES = 4;

    public static final SettingsState.ErrorTypeSettingsState ERROR = new SettingsState.ErrorTypeSettingsState(
            true,
            true, (new Color(40, 20, 0)).getRGB(),
            true, (new Color(200, 60, 60)).getRGB()
    );
    public static final SettingsState.ErrorTypeSettingsState WARNING = new SettingsState.ErrorTypeSettingsState(
            true,
            true, (new Color(60, 60, 0)).getRGB(),
            true, (new Color(200, 140, 40)).getRGB()
    );
    public static final SettingsState.ErrorTypeSettingsState WEAK_WARNING = new SettingsState.ErrorTypeSettingsState(
            true,
            true, (new Color(60, 30, 0)).getRGB(),
            true, (new Color(200, 200, 100)).getRGB()
    );
    public static final SettingsState.ErrorTypeSettingsState INFORMATION = new SettingsState.ErrorTypeSettingsState(
            true,
            true, (new Color(35, 35, 35)).getRGB(),
            true, (new Color(200, 200, 200)).getRGB()
    );
    public static final SettingsState.ErrorTypeSettingsState OTHER = new SettingsState.ErrorTypeSettingsState(
            true,
            true, (new Color(15, 10, 10)).getRGB(),
            true, (new Color(100, 100, 100)).getRGB()
    );

    public static final List<String> IGNORE_LIST = List.of("TYPO", "TODO", "is never used");
}

File: src/main/kotlin/com/github/ioannuwu/errorlens/appearance/DefaultSettingsList.kt
package com.github.ioannuwu.errorlens.appearance

import com.github.ioannuwu.errorlens.appearance.ErrorTypeList
import com.github.ioannuwu.errorlens.gui.Settings
import java.awt.Color

object DefaultSettingsList: ErrorTypeList<Settings> {

    val SPACES = 2

    val HIDE_LIST = ""

    override val ERROR: Settings = Settings(
            true,
            true, Color(40, 20, 0).rgb,
            true, Color(200, 60, 60).rgb)

    override val WARNING: Settings = Settings(
            true,
            true, Color(60, 60, 0).rgb,
            true, Color(200, 140, 40).rgb)

    override val WEAK_WARNING: Settings = Settings(
            false,
            true, Color(60, 30, 0).rgb,
            true, Color(200, 200, 100).rgb)

    override val INFORMATION: Settings = Settings(
            false,
            false, Color(35, 35, 35).rgb,
            false, Color(200, 200, 200).rgb)

    override val OTHER: Settings = Settings(
            false,
            false, Color(15, 10, 10).rgb,
            false, Color(100, 100, 100).rgb)
}